File: src/main/贪心算法/t861/matrixScore.go
package t861

func matrixScore(A [][]int) int {
	rows, cols := len(A), len(A[0])
	for i := 0; i < rows; i++ {
		if A[i][0] == 0 {
			for j := 0; j < cols; j++ {
				A[i][j] ^= 1
			}
		}
	}
	res := 0
	for i := 0; i < cols; i++ {
		count := 0
		for j := 0; j < rows; j++ {
			count += A[j][i]
		}
		if rows-count > count {
			count = rows - count
		}
		res += count * (1 << (cols - i - 1))
	}
	return res
}

File: src/main/贪心算法/t861/MatrixScore.java
package 贪心算法.t861;

/**
 * @author: sotowang
 * @date: 2020/12/07 15:39
 */
public class MatrixScore {
    public int matrixScore(int[][] A) {
        int rows = A.length;
        int cols = A[0].length;
        for(int i=0;i<rows;i++){
            if(A[i][0] == 0){
                for(int j=0;j<cols;j++){
                    A[i][j] ^= 1;
                }
            }
        }
        int res = 0;
        for(int i=0;i<cols;i++){
            //统计每一列有多少个1
            int count = 0;
            for(int j=0;j<rows;j++){
                count += A[j][i];
            }
            res += Math.max(rows-count,count)*(1<<cols-i-1);
        }
        return res;
    }
}
